### PR TITLE
Magician unit test

### PR DESCRIPTION
**Pull request by M0rrighan**
Added in this branch:
### unit tests for

- calculate.js
- operate.js

### CODE REVIEW

File: src/tests/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    const result = operate('5', '0', '÷');
+    expect(result).toBe("Can't divide by 0.");
+  });
+
+  test('any other operator throws an error message', () => {
+    expect(() => {
+      operate('5', '0', '/');
+    }).toThrow("Unknown operation '/'");
+  });
+});
```
Comment (Emmyn5600) [inline src/tests/operate.test.js:50]: ### Let us add the following tests.

- [ ] Should give the sum of two given numbers is the operator is +
- [ ] Should give the difference of two given numbers is the operator is -
- [ ] Should give the product of two given numbers is the operator is x
- [ ] Should give the percentage of the first number is the operator is %
- [ ] Should give the rest of the division is the operator is ÷
- [ ] Should throw an error if given a wrong number

File: src/tests/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    });
+  });
+
+  test('"+/-" operator returns signed number', () => {
+    objArr.forEach((obj, i) => {
+      const result = calculate(obj, '+/-');
+      expect(result.next).toEqual(signedArr[i]);
+    });
+  });
+});
```
Comment (Emmyn5600) [inline src/tests/calculate.test.js:53]: ### Let us add the following tests.

- [ ] should give the some of two given number is the operation is +
- [ ] should divide the two given numbers'
- [ ] should give the product of two given number is the operation is x
- [ ] should give the difference of two given number is the operation is -
- [ ] should give the percentage of the first operand if the button name is %
- [ ] should return 0 as next if the button name is AC
- [ ] should return 23. as next if the button name is .
- [ ] should throw 0 error if trying to make a zero division
- [ ] should throw Zero division error.

File: src/components/calculator.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Calculator from './Calculator';
+
+it('calculator snapshot test', () => {
+  const tree = renderer
+    .create(<Calculator />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (Emmyn5600) [inline src/components/calculator.test.js]: - [ ] Let us move some tests files inside the test folder that are inside the component.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @M0rrighan and @Mucyosoda 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.

 ## Required Changes ♻️

- _Check the comment under the review._

## Optional suggestions

- n/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/tests/operate.test.js (hunk 1 above)
Comment (Mucyosoda) [reply src/tests/operate.test.js:50]: Hello Emmyn5600, the result of each operator is in resultArr

**Review by ZeenatLawal — APPROVED** [review]
Hi @M0rrighan and @Mucyosoda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 
6vEo


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**